airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -0,0 +1,797 @@
+        nodegroup_name: Optional[str],
+        aws_conn_id: Optional[str] = DEFAULT_CONN_ID,
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.cluster_name = cluster_name
+        self.nodegroup_subnets = nodegroup_subnets
+        self.nodegroup_role_arn = nodegroup_role_arn
+        self.nodegroup_name = nodegroup_name or cluster_name + datetime.now().strftime("%Y%m%d_%H%M%S")
+        self.aws_conn_id = aws_conn_id
+        self.region = region
+
+    def execute(self, context):
+        eks_hook = EKSHook(
+            aws_conn_id=self.aws_conn_id,
+            region_name=self.region,
+        )
+
+        return eks_hook.create_nodegroup(
+            clusterName=self.cluster_name,
+            nodegroupName=self.nodegroup_name,
+            subnets=self.nodegroup_subnets,
+            nodeRole=self.nodegroup_role_arn,
+        )
+
+
+class EKSDeleteClusterOperator(BaseOperator):
+    """
+    Deletes the Amazon EKS Cluster control plane and all nodegroups attached to it.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:EKSDeleteClusterOperator`
+
+    :param cluster_name: The name of the Amazon EKS Cluster to delete.
+    :type cluster_name: str
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+         If this is None or empty then the default boto3 behaviour is used. If
+         running Airflow in a distributed manner and aws_conn_id is None or
+         empty, then the default boto3 configuration would be used (and must be
+         maintained on each worker node).
+    :type aws_conn_id: str
+        :param region: Which AWS region the connection should use.
+        If this is None or empty then the default boto3 behaviour is used.
+    :type region: str
+
+    """
+
+    template_fields = (
+        "cluster_name",
+        "aws_conn_id",
+        "region",
+    )
+
+    def __init__(
+        self,
+        cluster_name: str,
+        aws_conn_id: Optional[str] = DEFAULT_CONN_ID,
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.cluster_name = cluster_name
+        self.aws_conn_id = aws_conn_id
+        self.region = region
+
+    def execute(self, context):
+        eks_hook = EKSHook(
+            aws_conn_id=self.aws_conn_id,
+            region_name=self.region,
+        )
+
+        nodegroups = eks_hook.list_nodegroups(clusterName=self.cluster_name).get('nodegroups')
+        nodegroup_count = len(nodegroups)
+        if nodegroup_count > 0:
+            self.log.info(
+                "A cluster can not be deleted with attached nodegroups.  Deleting %d nodegroups.",
+                nodegroup_count,
+            )
+            for group in nodegroups:
+                eks_hook.delete_nodegroup(clusterName=self.cluster_name, nodegroupName=group)
+
```

Comment:
@mik-laj @ashb @o-nikolas  - I added this as a convenience but now I'm having second thoughts and want your opinion.  By default, the API will not allow you to delete a Cluster which has attached Nodegroups.  I added this block to check for nodegroups and delete them for you rather than throwing an exception, but I now wonder if I should put that behind a `force delete` flag rather than circumvent the system behavior by default?

Reply:
Yeah, you are probably right that having it behind a `force=True` flag would be better.

Reply:
Addressed in https://github.com/apache/airflow/pull/16571/commits/553e4835ea846c447f21398effefd0d1dbc13c4c